Feature/1.8/kqueue epoll

lazan opened the pull request.

lazan committed:
common: using kqueue for 'select' on BSD plattforms

lazan committed:
common: filename result.h -> posix.h

namespace in `result.h` was `common::posix. `posix.h` is a better name for
content.

lazan committed:
common: using epoll on linux for multiplexing

This patch uses epoll on linux. `epoll` should be a lot more efficient than
`pselect`. Otherwise there's no semantic changes.

resolves #89

lazan committed:
unittest: glitch in unittest in gateway_protocol_1_0_manager

`domain_A__connect_as_inbound__call_x___expect_correct_call_request` did
not wait for outbound to register discovery ability.

lazan committed:
file: missing cmk install directive

KristianIvarsson commented on middleware/common/include/common/communication/select/epoll.h:
```diff
@@ -0,0 +1,163 @@
+      namespace directive
+      {
+ 
+         using range_type = range::type_t< std::vector< strong::file::descriptor::id>>;
```
`type_t` has to be a tautologi because `t` is an abbreviation for `type`

lazan replied:
Yeah... I tried to mimic `std` _helper types_.  Such as `std::remove_reference` has a _helper type_ `std::remove_reference_t`

KristianIvarsson commented on middleware/common/include/common/communication/select/epoll.h:
```diff
@@ -0,0 +1,163 @@
+
+            template< typename Range>
+            auto remove( Range&& descriptors) noexcept
+               -> decltype( remove( range::front( descriptors)))
```
There's no return  so having trailing return type declaration seems to be ambigious

lazan replied:
I'm using trailing return type with _expresion template_ (i think it's called) to make sure that elements in the `Range` can be used. If not `decltype( remove( range::front( descriptors)))` evaluates to a type (void in this case) the whole function is not instantiated.
Like `enable_if`, but I think this _technique_ is much cleaner and easier to read.

KristianIvarsson replied:
Apparently it is not easier to read if the purpose was misunderstood ;-)

lazan replied:
I think you'll prefer this over enable_if boilerplate. 
But, now we could use ad-hoc `requires` which might be more informative.

lazan replied:
Apparently I updated it to concepts.

lazan committed:
build: removed casual-common-serialize-service lib -> regular common

This batch removes the lib `casual-common-serialize-service` and just
link the serialize-service TU:s to casual-common.

Also: made casual-manager an archive, since this is only linked in from
executables, never from a lib.
